presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoResultSet.java

```diff
@@ -1816,6 +1819,37 @@ private ResultsPageIterator(StatementClient client, Consumer<QueryStats> progres
         }
     }
 
+    private static class LengthLimitedIterator<T>
```

Comment:
Add a comment that this is like Guava's `Iterators.limit()`, but uses a `long` limit instead of `int` (since my first thought reading this is that Guava has the same thing)